assets/app/view/train_roster.rb

```diff
@@ -27,32 +27,115 @@ def render_body
         ])
       end
 
+      children << phases
+
       h(:div, {}, children)
     end
 
+    def cert_limit
+      cert_props = {
+        style: {
+          'margin-bottom': '1rem'
+        }
+      }
+      h(:div, cert_props, "Certificate limit: #{@game.cert_limit}")
+    end
+
+    def phases
+      td_props = {
+        style: {
+          padding: '0 1rem'
+        }
+      }
+
+      current_phase = @game.phase.current
+      rows = @game.phase.phases.map do |phase|
+        phase_color = Array(phase[:tiles]).last
+        phase_props = {
+          style: {
+            padding: '0 1rem'
+          }
+        }
+        if View::Part::MultiRevenue::COLOR.include? phase_color
+          m = View::Part::MultiRevenue::COLOR[phase_color].match(/#(..)(..)(..)/)
```

Comment:
i've seen this pattern before, can this be refactored?

Reply:
Done